projects/cm_mcu/CommandLineTask.c

```diff
@@ -56,41 +56,33 @@ static BaseType_t bl_ctl(int argc, char **argv, char *m)
   return pdFALSE;
 }
 
+#ifdef REV2
 // this command takes one argument
-static BaseType_t clock_ctl(int argc, char **argv, char *m)
+static BaseType_t clearclk_ctl(int argc, char **argv, char *m)
 {
   int copied = 0;
   int status = -1; // shut up clang compiler warning
-  BaseType_t i = strtol(argv[1], NULL, 10);
-  if (!((i == 1) || (i == 2))) {
-    copied +=
-        snprintf(m + copied, SCRATCH_SIZE - copied,
-                 "Invalid mode %ld for clock, only 1 (reset) and 2 (program) supported\r\n", i);
+
+  // acquire the semaphore
+  if (acquireI2CSemaphore(i2c2_sem) == pdFAIL) {
+    snprintf(m + copied, SCRATCH_SIZE - copied, "%s: could not get semaphore in time\r\n", argv[0]);
     return pdFALSE;
   }
-  copied += snprintf(m + copied, SCRATCH_SIZE - copied, "%s mode set to %ld. \r\n", argv[0], i);
-  if (i == 1) {
-    status = initialize_clock();
-    if (status == 0)
-      copied += snprintf(m + copied, SCRATCH_SIZE - copied,
-                         "clock synthesizer successfully initialized. \r\n");
+  status = clear_clk_stickybits();
+  // check if we have the semaphore
+  if (xSemaphoreGetMutexHolder(i2c2_sem) == xTaskGetCurrentTaskHandle()) {
+    xSemaphoreGive(i2c2_sem);
   }
-  else if (i == 2) {
-    status = load_clock();
-    if (status == 0)
-      copied += snprintf(m + copied, SCRATCH_SIZE - copied,
-                         "clock synthesizer successfully programmed. \r\n");
+
+  if (status == 0)
+    snprintf(m + copied, SCRATCH_SIZE - copied,
+             "clock synthesizer successfully clear sticky bits. \r\n");
+  else {
+    snprintf(m + copied, SCRATCH_SIZE - copied, "%s operation failed \r\n", argv[0]);
```

Comment:
let's just print something on failure, and assume success.

Reply:
Done.